test/test_ops.py

```diff
@@ -231,6 +238,36 @@ def test_reference_testing(self, device, dtype, op):
         finally:
             torch.set_default_dtype(cur_default)
 
+    @unittest.skipIf(TEST_WITH_ASAN, "Skipped under ASAN")
+    @onlyNativeDeviceTypes
+    @skipIfCrossRef
+    @suppress_warnings
+    @ops(_decomp_test_ops, allowed_dtypes=(torch.float64, torch.long, torch.complex128))
+    def test_decomp(self, device, dtype, op):
+        hit = False
+
+        class DecompMode(torch.Tensor):  # TODO: don't subclass from tensor
+            @classmethod
+            def __torch_dispatch__(cls, func, types, args, kwargs):
+                # TODO: this no_dispatch call shouldn't be necessary after
+                # https://github.com/pytorch/pytorch/pull/75965
+                with no_dispatch():
+                    if func in decomposition_table:
+                        nonlocal hit
+                        hit = True
+                        return decomposition_table[func](*args, **kwargs)
+                    else:
+                        return func(*args, **kwargs)
+
```

Comment:
@Chillee wrote tests for decompositions over at https://github.com/pytorch/functorch/blob/955decd01aa7fc391f2bfbbb7ae5de0f9ed8ed0a/test/test_ops.py#L1349, should we just upstream those? These cover autograd testing and some additional dtypes.